elegant/clean_timepoint_data.py

```diff
@@ -0,0 +1,147 @@
+        timepoint = '2018-07-30t1200'
+        remove_timepoint_for_position(experiment_root, position, timepoint)
+    """
+
+    experiment_root = pathlib.Path(experiment_root)
+    positions = [metadata_file.parent for metadata_file in sorted(experiment_root.glob('*/position_metadata.json'))]
+
+    for position in positions:
+        remove_timepoint_for_position(experiment_root, position, timepoint, dry_run=dry_run)
+
+    md_file = (experiment_root/'experiment_metadata.json')
+    with md_file.open() as md_fp:
+        expt_md = json.load(md_fp)
+
+    try:
+        timepoint_idx = expt_md['timepoints'].index(timepoint)
+    except ValueError:
+        print('Timepoint not found in experiment_metadata')
+        return
+
+    for list_entry_type in ['durations','timestamps','timepoints']:
+        del expt_md[list_entry_type][timepoint_idx]
+
+    for dict_entry_type in ['brightfield metering', 'fluorescent metering', 'humidity', 'temperature']:
+        expt_md[dict_entry_type] = {key:val for key,val in expt_md[dict_entry_type].items() if key != timepoint}
+
+    datafile.json_encode_atomic_legible_to_file(expt_md,experiment_root/'experiment_metadata.json')
+
+def remove_excluded_positions(experiment_root,dry_run=False):
+    """Deletes excluded positions from an experiment directory
+
+    Parameters
+        experiment_root - str/pathlib.Path to experiment
+        dry_run - bool flag that toggles taking action (if False, only specifies when offending files are found)
+    """
+
+    experiment_root = pathlib.Path(experiment_root)
+    annotations = load_data.read_annotations(experiment_root)
+    good_annotations = load_data.filter_annotations(annotations, load_data.filter_excluded)
+    excluded_positions = sorted(set(annotations.keys()).difference(set(good_annotations.keys())))
+    for position in excluded_positions:
+        if (experiment_root / position).exists():
+            print(f'Found an excluded position to delete {position}')
+            if not dry_run:
+                shutil.rmtree(str(experiment_root / position))
+
+def remove_dead_timepoints(experiment_root, postmortem_timepoints, dry_run=False):
+    """Deletes excess timepoints in an experiment where worms are dead
+
+    Parameters
+        experiment_root - str/pathlib.Path to experiment
+        postmortem_time - Number of timepoints to keep past the annotated death timepoint;
+            useful for keeping extra timepoints in case one ever wants to validate
+            death for the previously made annotations; postmortem_timepoints should be an int >= 1
+        dry_run - bool flag that toggles taking action (if False, only specifies when offending files are found);
+            this flag is exposed for corresponding functionality through remove_timepoint_for_position
+    """
+
+    if postmortem_timepoints < 1:
+        raise ValueError('postmortem_timepoints should be >= 1')
+    elif type(postmortem_timepoints) is not int:
```

Comment:
Is the type-check actually necessary? If so, use `isinstance(postmortem_timepoints, int)`.